builtin/plugins/coin/coin.go

```diff
@@ -213,6 +242,27 @@ func mint(ctx contract.Context, to loom.Address, amount *loom.BigUInt) error {
 	return ctx.Set(economyKey, econ)
 }
 
+//Mint : to be called by CoinDeflationManager Responsible to mint coins as per various parameter defined
+func Mint(ctx contract.Context) error {
+	var policy Policy
+	err := ctx.Get(policyKey, &policy)
+	if err != nil {
+		return errUtil.Wrap(err, "Failed to Get PolicyInfo")
+	}
+	var depreciation *common.BigUInt
+	policyNumerator := loom.NewBigUIntFromInt(int64(policy.DeflationFactorNumerator))
+	policyDenominator := loom.NewBigUIntFromInt(int64(policy.DeflationFactorDenominator))
+	blockHeight := loom.NewBigUIntFromInt(ctx.Block().Height)
+	depreciation = policyNumerator.Mul(policyNumerator, blockHeight)
+	depreciation.Div(depreciation, policyDenominator)
+	amount := loom.NewBigUIntFromInt(int64(policy.BaseMintingAmount))
+	if amount.Cmp(depreciation) <= 0 {
+		return nil // No more coins to be minted on block creation
+	}
+	amount.Sub(amount, depreciation)
```

Comment:
So the original issues https://github.com/loomnetwork/loomchain/issues/1182 says:
> someone could configure their chain to have 10% inflation first year, 5% second. Etc . Ideally they just configure the windows of time for rebalance.

This code doesn't do that.
Also, here the depreciation starts from the first block, which doesn't seem quite right, it should start applying once this feature is enabled (from the height at which it's enabled).

Reply:
Done.